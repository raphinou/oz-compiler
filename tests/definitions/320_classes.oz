% private and dynamic method labels
% also tests otherwise method
local
 C L A I
in
   A=echo
   class C
      meth init skip end
      meth !A(M)
         {Show M}
      end
      meth B(M)
         {Show private}
         {Show M}
      end
      meth whisper(M)
         {self B(M)}
      end
      meth otherwise(M)
         {Show otherwise}
         {Show M}
      end
   end
   I={New C init}
   {I echo(test)}
   {I whisper(shhhh)}
   {I bla(hehe)}
end

