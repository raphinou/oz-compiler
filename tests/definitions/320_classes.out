test
private
shhhh
otherwise
bla(hehe)
